miden-lib/src/notes/mod.rs

```diff
@@ -2,11 +2,64 @@ use crate::memory::{
     CREATED_NOTE_ASSETS_OFFSET, CREATED_NOTE_CORE_DATA_SIZE, CREATED_NOTE_HASH_OFFSET,
     CREATED_NOTE_METADATA_OFFSET, CREATED_NOTE_RECIPIENT_OFFSET, CREATED_NOTE_VAULT_HASH_OFFSET,
 };
-use crypto::{hash::rpo::RpoDigest as Digest, StarkField, Word, WORD_SIZE};
+use assembly::ast::ProgramAst;
+use crypto::{hash::rpo::RpoDigest as Digest, StarkField, Word, WORD_SIZE, Felt};
 use miden_objects::{
-    notes::{NoteMetadata, NoteStub, NoteVault},
-    NoteError,
+    notes::{NoteMetadata, NoteStub, NoteVault, Note, NoteScript},
+    NoteError, accounts::AccountId, assets::Asset,
 };
+use crate::assembler::assembler;
+
+pub enum Script {
+    P2ID { target: AccountId },
+    P2IDR {
+        target: AccountId,
+        recall_height: u32,
+    },
+}
+
+pub fn create_note_with_script(
+    script: Script,
+    assets: Vec<Asset>,
+    sender: AccountId,
+    tag: Option<Felt>,
+    serial_num: Word,
+) -> Result<Note, NoteError> {
+
+    let mut note_assembler = assembler();
+    let (note_script, inputs): (&str, Vec<Felt>) = match script {
+        Script::P2ID { target } => ("p2id", vec![target.into()]), // Convert `target` to a suitable type if necessary
+        Script::P2IDR { target, recall_height } => ("p2idr", vec![target.into(), recall_height.into()]), // Convert both to a suitable type
+    };
+
+    // Create the note
+    let note_script_ast = ProgramAst::parse(
+        format!(
+            "
+        use.miden::note_scripts::basic
+    
+        begin
+            exec.basic::{note_script}
+        end
```

Comment:
Ideally, this would be replaced by just reading contents of a file (after https://github.com/0xPolygonMiden/miden-vm/issues/1098 is implemented).